### ISSUE TITLE
BUG h5py does not let me save the file anymore

### ISSUE THREAD

**Issue by lukyluket** [tenpy/tenpy#125]
**Describe the bug**
I am trying to save the data of my simulations. Everything work on my laptop with install python 3.8.5 tenpy 0.2.7 and h5py 2.10.0

On my desktop I have reinstall everything from scracth and just used the 
conda install --channel=conda-forge physics-tenpy
conda create --name tenpy --channel=conda-forge physics-tenpy
conda activate tenpy

As a result I end up with 
python 3.9.2
GCC 9.3.0
tenpy 0.7.2
h5py 3.1.0

As mentioned I cannot read the files generated by the laptop because of some boundary conditions type error, so I decided to regenerate all my data.  But as soon as I try to write the data to the disk I get the error.

The file is a minimal instance of DMRG,

       
  
   
   
     def find_ground_state(L,theta0,chi,dmrg_params):
    model0 = Potts_symm({"L": L, "J": - np.sin(theta0), "f": - np.cos(theta0), "bc_MPS": "finite"})
    sites = model0.lat.mps_sites()
    psi = MPS.from_product_state(sites, ['0'] * (L), "finite")
    dmrg.run(psi, model0, dmrg_params)
    return psi
    def main(argv):   
     """
    Arguments: L, theta0=np.pi/arg ,max_chi_gs, max_chi    
    """
    data_path = 'Data/' 
    name_file = 'ground_state'
    #L = 128  # number of sites
    L = int(argv[1])
    theta = np.pi/float(argv[2])
    #theta0 = np.arctan(1./4.1);
    #theta = np.pi/4;
    chi = int(argv[3])
    model = Potts_symm({"L": L, "J": - np.sin(theta), "f": - np.cos(theta), "bc_MPS": "finite"})  
    dmrg_params = {
        'trunc_params': {
        'chi_max': chi,
        'svd_min': 1.e-10,
         #'trunc_cut': None
        },
        'conserve':'z3',
        'update_env': 20,
        'start_env': 20,
        'max_E_err': 1.e-10,
        'max_S_err': 1.e-10,
        'verbose': 1,
        'mixer': False
    }
    psi = find_ground_state(L,theta,chi,dmrg_params)
    max_chi = np.max(psi.chi)
    data = {"psi": psi,  # e.g. an MPS
        "model": model,
        "dmrg_params": dmrg_params,
        "parameters": {"L": L, "theta": theta},
         "chi_t": max_chi
         }
    name_file = data_path+name_file + '_L_' + str(L) + '_theta_' +str(theta)+\
     '_max_chi_'+str(max_chi)+'_symm_'
    with h5py.File(name_file+'.h5', 'w') as f:
        hdf5_io.save_to_hdf5(f, data)

[Traceback]
```
  File "/home/luca/potts_3/Python_Code/find_ground_state.py", line 76, in <module>
    main(sys.argv)     
  File "/home/luca/potts_3/Python_Code/find_ground_state.py", line 74, in main
    hdf5_io.save_to_hdf5(f, data)
  File "/home/luca/anaconda3/envs/tenpy/lib/python3.9/site-packages/tenpy/tools/hdf5_io.py", line 926, in save_to_hdf5
    return Hdf5Saver(h5group).save(obj, path)
  File "/home/luca/anaconda3/envs/tenpy/lib/python3.9/site-packages/tenpy/tools/hdf5_io.py", line 410, in save
    h5gr = f(self, obj, path, type_repr)
  File "/home/luca/anaconda3/envs/tenpy/lib/python3.9/site-packages/tenpy/tools/hdf5_io.py", line 539, in save_dict
    type_repr = self.save_dict_content(obj, h5gr, subpath)
  File "/home/luca/anaconda3/envs/tenpy/lib/python3.9/site-packages/tenpy/tools/hdf5_io.py", line 577, in save_dict_content
    self.save(v, subpath + k)
  File "/home/luca/anaconda3/envs/tenpy/lib/python3.9/site-packages/tenpy/tools/hdf5_io.py", line 422, in save
    obj_save_hdf5(self, h5gr, subpath)  # should save the actual data
  File "/home/luca/anaconda3/envs/tenpy/lib/python3.9/site-packages/tenpy/networks/mps.py", line 265, in save_hdf5
    hdf5_saver.save(self.sites, subpath + "sites")
  File "/home/luca/anaconda3/envs/tenpy/lib/python3.9/site-packages/tenpy/tools/hdf5_io.py", line 410, in save
    h5gr = f(self, obj, path, type_repr)
  File "/home/luca/anaconda3/envs/tenpy/lib/python3.9/site-packages/tenpy/tools/hdf5_io.py", line 513, in save_iterable
    self.save_iterable_content(obj, h5gr, subpath)
  File "/home/luca/anaconda3/envs/tenpy/lib/python3.9/site-packages/tenpy/tools/hdf5_io.py", line 534, in save_iterable_content
    self.save(elem, subpath + str(i))
  File "/home/luca/anaconda3/envs/tenpy/lib/python3.9/site-packages/tenpy/tools/hdf5_io.py", line 422, in save
    obj_save_hdf5(self, h5gr, subpath)  # should save the actual data
  File "/home/luca/anaconda3/envs/tenpy/lib/python3.9/site-packages/tenpy/tools/hdf5_io.py", line 266, in save_hdf5
    type_repr = hdf5_saver.save_dict_content(self.__dict__, h5gr, subpath)
  File "/home/luca/anaconda3/envs/tenpy/lib/python3.9/site-packages/tenpy/tools/hdf5_io.py", line 577, in save_dict_content
    self.save(v, subpath + k)
  File "/home/luca/anaconda3/envs/tenpy/lib/python3.9/site-packages/tenpy/tools/hdf5_io.py", line 422, in save
    obj_save_hdf5(self, h5gr, subpath)  # should save the actual data
  File "/home/luca/anaconda3/envs/tenpy/lib/python3.9/site-packages/tenpy/linalg/np_conserved.py", line 318, in save_hdf5
    hdf5_saver.save(self.dtype, subpath + "dtype")
  File "/home/luca/anaconda3/envs/tenpy/lib/python3.9/site-packages/tenpy/tools/hdf5_io.py", line 427, in save
    raise Hdf5ExportError(msg)
tenpy.tools.hdf5_io.Hdf5ExportError: Don't know how to save object of type <class 'numpy.dtype[float64]'>:
dtype('float64')

**Comment by jhauschild** [comment]
I can reproduce your issue. A minimal example:
```python
import numpy as np
#import hdf5_io
from tenpy.tools import hdf5_io

data = {'dtype': np.dtype('float64')}
hdf5_io.save(data, 'test.h5')
```
It turns out the issue is coming from the new numpy version 1.20 that you probably installed alongside in the new environment,
see https://numpy.org/doc/stable/release/1.20.0-notes.html#isinstance-dtype-np-dtype-and-not-type-dtype-is-not-np-dtype
So as a first  quick workaround, you can simply install a older numpy version, e.g. 1.19.

**Comment by lukyluket** [comment]
You are right, downgrading numpy solves the issue. Thank you very much

**Comment by jhauschild** [comment]
Okay, I fixed this in daf35cfcaf0fbd9c0d333efa82d8e9e847e1d76d
It should now work with numpy version 1.20 as well. Lot's of deprecation warnings pop up, but that's a separate issue #127
